src/Provider/DataSourceProvider.js

```diff
@@ -9,6 +9,33 @@ const error = (err, source) => {
     throw err;
 };
 
+function convertSourceData(data, extDest, layer) {
+    // Add missing parent style
+    if (data.isFeatureCollection) {
+        data.features.forEach((f) => {
+            let parentStyle = layer.style;
+
+            if (f.style && f.style.isStyle) {
+                parentStyle = f.style;
+
+                if (layer.style && !f.style.parent.isStyle) {
```

Comment:
you check if `style.parent` is defined to avoid reassigning?

Reply:
Yes, it was to set it only once if missing.